libs/utils/android/benchmark.py

```diff
@@ -187,4 +193,70 @@ def __init__(self):
         self.log.info('=== Finalization...')
         self.benchmarkFinalize()
 
+    def _adb(self, cmd):
+        os.system('adb -s {} {}'.format(self.target.adb_name, cmd))
+
+    def _fastboot(self, cmd):
+        os.system('fastboot -s {} {}'.format(self.target.adb_name, cmd))
+
+
+    def _wait_for_logcat_idle(self, seconds=1):
+        lines = 0
+
+        # Clear logcat
+        # os.system('{} logcat -s {} -c'.format(adb, DEVICE));
+        self._adb('logcat -c')
+
+        # Dump logcat output
+        logcat_cmd = 'adb -s {} logcat'.format(self.target.adb_name)
+        logcat = Popen(logcat_cmd, shell=True, stdout=PIPE)
+        logcat_poll = select.poll()
+        logcat_poll.register(logcat.stdout, select.POLLIN)
+
+        # Monitor logcat until it's idle for the specified number of [s]
+        self.log.info('Waiting for system to be almost idle')
+        self.log.info('   i.e. at least %d[s] of no logcat messages', seconds)
+        while True:
+            poll_result = logcat_poll.poll(seconds * 1000)
+            if not poll_result:
+                break
+            lines = lines + 1
+            line = logcat.stdout.readline(1024)
+
+    def _reboot(self):
+
+        # Reboot the device, if a boot_image has been specified
+        if self.args.boot_image:
+
+            self.log.warning('=== Rebooting...')
+            self.log.warning('Rebooting image to use: %s', self.args.boot_image)
+
+            self.log.debug('Waiting 6[s] to enter bootloader...')
+            self._adb('reboot-bootloader')
+            sleep(6)
+            self._fastboot('boot {}'.format(self.args.boot_image))
+
+            self.log.debug('Waiting 20[s] for boot to start...')
+            sleep(20)
```

Comment:
Before I forget, one more comment:
The sleep value here has to be higher as the boot process takes longer than 20s. I would recommend 60s.